src/psyclone/psyir/frontend/fparser2.py

```diff
@@ -3965,6 +3966,14 @@ def _case_construct_handler(self, node, parent):
             if isinstance(child, Fortran2003.End_Select_Stmt):
                 clause_indices.append(idx)
 
+        # Deeply-nested ifblocks are problematic because they will exceed the
+        # python recursion limits for some psyir analysis. These are not
+        # typically found in input code, but select-case with a large number
+        # of cases are more probable. If we find one of such cases we
+        # preventively increase the python recursion limits.
+        if len(clause_indices) > 150:
```

Comment:
Please add a TODO for #11 as it would be good to log this.

Reply:
Done